scenarios/pfcp-app-filtering.xml

```diff
@@ -3,56 +3,104 @@
  ~ SPDX-FileCopyrightText: 2022-present Open Networking Foundation <info@opennetworking.org>
   -->
 <scenario name="pfcp-app-filtering" description="Set up and test application filtering with the PFCP client">
+<!-- FIXME: create two app with ALLOW and DENY to test app filtering in the same session.
+      Currently we do two separate tests one with ALLOW and another with DENY because
+      pfcpsim doesn't support multiple app-filtering rules
+-->
     <group name="Pfcp-App-Filtering">
-        <group name="Pfcp-Setup-App-Filtering">
-            <group name="Pfcp-Push-App-Filtering" delay="5">
+        <group name="Pfcp-Setup-App-Filtering-1">
+            <group name="Pfcp-Push-App-Filtering-1" delay="5">
                 <sequential var="${ENODEB#}"
-                            starts="Pfcp-Session-Create-App-Filtering-${#}"
-                            ends="Pfcp-Session-App-Filtering-${#-1}">
-                    <step name="Pfcp-Session-Create-App-Filtering-${#}"
-                          exec="${DOCKER_COMPOSE_CMD} exec -T mock-smf pfcpctl session create --count 5 --baseID ${#}0 --ue-pool 17.0.${#-1}.0/24 --gnb-addr 140.0.10${#-1}.1 --sdf-filter 'permit out ip from 0.0.0.0/0 to assigned 81-81'"/>
-                    <step name="Pfcp-Session-App-Filtering-${#}" requires="^"
+                            starts="Pfcp-Session-Create-App-Filtering-${#}-1"
+                            ends="Pfcp-Session-App-Filtering-${#-1}-1">
+                    <step name="Pfcp-Session-Create-App-Filtering-${#}-1"
+                          exec="${DOCKER_COMPOSE_CMD} exec -T mock-smf pfcpctl session create --count 5 --baseID ${#}0 --ue-pool 17.0.${#-1}.0/24 --gnb-addr 140.0.10${#-1}.1 --app-filter 'udp:0.0.0.0/0:81-81:allow'"/>
+                    <step name="Pfcp-Session-App-Filtering-${#}-1" requires="^"
                           exec="${DOCKER_COMPOSE_CMD} exec -T mock-smf pfcpctl session modify --count 5 --baseID ${#}0 --ue-pool 17.0.${#-1}.0/24 --gnb-addr 140.0.10${#-1}.1"/>
                 </sequential>
             </group>
-            <step name="Check-Up4-Flows-App-Filtering" requires="Pfcp-Push-App-Filtering" delay="5"
+            <step name="Check-Up4-Flows-App-Filtering-1" requires="Pfcp-Push-App-Filtering-1" delay="5"
                   exec="onos-cli-grep ${OCI} up4:read-flows 'Apps=1, UL sess=${PFCP_UP4_FLOWS}, UL flows=${PFCP_UP4_FLOWS}, DL flows=${PFCP_UP4_FLOWS}'"/>
-            <step name="Check-Flow-Rules-App-Filtering" requires="Pfcp-Push-App-Filtering" delay="5"
+            <step name="Check-Flow-Rules-App-Filtering-1" requires="Pfcp-Push-App-Filtering-1" delay="5"
                   exec="onos-check-flows ${OCI}"/>
         </group>
-        <group name="Check-Traffic-App-Filtering-Negative" requires="Pfcp-Setup-App-Filtering" delay="5">
+        <group name="Check-Traffic-App-Filtering-Negative-1" requires="Pfcp-Setup-App-Filtering-1" delay="5">
```

Comment:
I am confuse here - we are not setting any gating - why is it getting dropped ?

Reply:
I hardcoded the port and added a comment explaining that we do a negative test for traffic not matching the application filter, that should be dropped.